### ISSUE TITLE
Training on a large dataset does not fit

### ISSUE THREAD

**Issue by TomAvrech** [Project-MONAI/research-contributions#121]
**Describe the bug**
Network "not training" with a large custom dataset.
Training and validation losses are equal after 1/2 epoches.

**To Reproduce**
Using a custom dataset generator, I produced 25k+ samples (unlike the very few MRI/etc samples used in this paper).
A sample is made of 6 channels of (48, 48, 48) tensors, and the output is 2 channels of (48, 48, 48) tensors.

The input is a 3D voxels representation of the "world". 
The output is a 3D voxels representation of the action needed to be executed according to the input.
dtype is np.float32, and the 2 output tensors are complementary to each other - If channel 0 contains 1.0 in some coordinates, in channel 1 at the same location the value will be: (1.0 - x), or 0.0 in this case.
This is like segmentation for 3D space.... a voxel can be "full or empty" (channel 0 = full, channel 1 = empty).

I'm using the DiceCELoss, and generally trying to go along with this paper's guidance.

Overfitting a single sample - works well.
I use an "aggressive" learning rate of 0.01, 1k+- epochs, print the voxels results, and it looks good - like the ground truth label.

When I train using the 25k dataset, both the training and validation losses get stuck and equal after 1/2 epochs.
I try different learning rates, and LR decay approaches:
Start at 0.01     and decay to 0.00001
Start at 0.0001 and decay to 0.00001

I also tried to use different (smaller) patch sizes: 16, 8, and 4.

**Expected behavior**
Train well on a large dataset. Not only 1 data sample overfitting, or small datasets (like in the paper - MRI ones).

**Environment (please complete the following information):**
 - OS: Windows 
 - Python version 3.7.7
 - MONAI version [e.g. git commit hash] 1.0
 - CUDA/cuDNN version  cudnn 8.2.1.32
 - GPU models and configuration 2080 / 2080Ti
